### PR TITLE
SWPBL-188127 -- Window management and block info display

### PR DESCRIPTION
**Pull request by jehan-diaz**
When a user clicks on a block, the list of attributes and their values for that block type must be displayed. These attributes, which for now include pre-set and standardized attributes for each block, can be seen in the multi-purpose panel when hovering over a block.

Right now, we are able to see two panels in the same main app window view at the same time. We are now also able to scan over the enter central editor using the minimap, which allows you to select a node to focus into the center of the editor.

### CODE REVIEW

File: example/Dynamite/editor.h [hunk 1]
```diff
@@ -2,27 +2,15 @@
...
-        //ImNodes::EditorContextSet(context);
-    }
-
-    /*~Editor() {
-        ImNodes::EditorContextFree(context);
-    } */
-
-    static void init() {
-
+    static void Editor::init() {
```
Comment (EvanShenkman-Sonos) [inline example/Dynamite/editor.h:13]: This should be `static void init()`

I'm surprised that this compiles for you. Are you using Visual Studio?

File: example/Dynamite/editor.h [hunk 2]
```diff
@@ -34,26 +22,14 @@ class Editor {
...
-            //context.addBlock();
-        }
-        //context.displayInEditor();
-    */
-        ImNodes::EndNodeEditor();
-    }
-
-    static void exit(ImNodesEditorContext* context) {
-        ImNodes::EditorContextFree(context);
+    static void Editor::exit() {
```
Comment (EvanShenkman-Sonos) [inline example/Dynamite/editor.h:25]: Same comment here, drop `Editor::`

File: example/Dynamite/multipanel.cpp [hunk 3]
```diff
@@ -1,8 +1,6 @@
-#include "multipanel.h"
-#include <imgui.h>
-#include <vector>
-#include <string>
+#pragma once
```
Comment (EvanShenkman-Sonos) [inline example/Dynamite/multipanel.cpp:1]: Delete this line. You don't need this in a source file.

**Review by EvanShenkman-Sonos — CHANGES_REQUESTED** [review]
Couldn't get it to compile. Make some small quick fixes and I'll take a closer look once I can build it.

File: example/Dynamite/editor.h (hunk 1 above)
Comment (jehan-diaz) [reply example/Dynamite/editor.h]: Unclear but I think it has to do with what version of C++ Visual Studio (not VS Code, which is what I and Kevin) uses. Visual Studio is apparently Microsoft specific

**Comment by KevinKimSonos** [comment]
Seeing a weird behavior where if I create multiple nodes and right click one of the nodes, I see multiple "Bypass" and "Other options
Screen Shot 2022-07-26 at 10 15 39 AM
"

**Comment by EvanShenkman-Sonos** [comment]
> Seeing a weird behavior where if I create multiple nodes and right click one of the nodes, I see multiple "Bypass" and "Other options Screen Shot 2022-07-26 at 10 15 39 AM "

Two things here:

1. We're generally _**not**_ allowed to upload images to GitHub. It's not a huge deal for this project, however GitHub stores the images publicly and it's an easy way for confidential info to leak.

2. I don't see this behavior. I've added many blocks and right clicking on each of them only show Bypass and Other.

**Review by EvanShenkman-Sonos — APPROVED** [review]

**Comment by KevinKimSonos** [comment]
> > Seeing a weird behavior where if I create multiple nodes and right click one of the nodes, I see multiple "Bypass" and "Other options Screen Shot 2022-07-26 at 10 15 39 AM "
> 
> Two things here:
> 
> 1. We're generally _**not**_ allowed to upload images to GitHub. It's not a huge deal for this project, however GitHub stores the images publicly and it's an easy way for confidential info to leak.
> 2. I don't see this behavior. I've added many blocks and right clicking on each of them only show Bypass and Other.

Thanks for letting me know about this. I will not upload images to GitHub in the future.

**Pull request merged**